lblod/worship-submissions-email-notification-service
# Bundling fixes

# Description
DL-5178

This PR is about fixing core features.

# Type of change


# Related services

- https://github.com/redpencilio/deliver-email-service/

# How to test

1. Add the following services to your app-worship-decisions-database `docker-compose.override.yml`
```yml
  deliver-email-service:
    image: redpencil/deliver-email-service:0.2.0
    environment:
      MAILBOX_URI: "http://data.lblod.info/id/mailboxes/1"
      WELL_KNOWN_SERVICE: "test"
  worship-submissions-email-notification-service:
    image: poltergeistz/worship-submissions-email-notification-service-local:latest
    environment:
      RUN_INTERVAL: 5
      OUTBOX_FOLDER_URI: "http://data.lblod.info/id/mail-folders/2"
      FROM_EMAIL_ADDRESS: "Agentschap Binnenlands Bestuur Vlaanderen <noreply-binnenland@vlaanderen.be>"
      WORSHIP_DECISIONS_APP_BASEURL: "https://databankerediensten.lokaalbestuur.vlaanderen.be/"
  frontend:
    image: poltergeistz/frontend-worship-decisions-local:latest # this is from https://github.com/lblod/frontend-worship-decisions/pull/9
```
2. Make sure you are on branch `feature/melding-functionality` in app-worship-decisions-database
3. Start the stack and pick a worship-service
4. Subscribe with some random email by clicking on "Ontvang mails bij nieuwe inzendingen"
5. Log for worship-submissions-email-notification-service to check if the task is running and the email is inserted
6. Log for deliver-email-service, use postman or thunderclient to make a POST request to `/email-delivery/` (Don't forget to set WELL_KNOWN_SERVICE: "test" in the deliver-email-service)
7. You should get an etheral link for the email preview
8. Next cronJob shouldn't find any submission 

---------------------------------------------------

If you want to go further you can sync loket and worship-decisions-database - see [how to sync](https://github.com/lblod/app-digitaal-loket/wiki/Extending-Delta-Producer)
1. Follow the first section of testing
2. Create and send a new worship-submission in loket with the same worship-unit
3. Let it produce and check if it's consumed in app-worship-decisions-database
4. Check the logs of worship-submissions-email-notification-service, it should find that submission
5. Same as first section "6." til "8."

Don't forget to check readme

# What to check

- Check if emails are inserted correctly.
- First subscription should sent a batch of submissions then only new ones when available.

# Links to other PR's

- https://github.com/lblod/app-worship-decisions-database/pull/41
- https://github.com/lblod/frontend-worship-decisions/pull/9

# Notes

**To simplify testing I've made images from the services so you don't have to build it**
